src/index.ts

```diff
@@ -1,7 +1,79 @@
+import { euclidean } from 'ml-distance-euclidean';
 /**
- * My module
- * @returns A very important number
+ * DBSCAN
+ * @returns list of cluster，-1 is noise
  */
-export function myModule(): number {
-  return 42;
+interface IPoint {
+  index: number;
+  value: number[];
+  label: number;
+}
+function rangeQuery(
+  cur: IPoint,
+  data: IPoint[],
+  epsilon: number,
+  distance: (p: number[], q: number[]) => number,
+) {
+  return data.filter((point) => distance(point.value, cur.value) <= epsilon);
+}
+export function dbscan(
```

Comment:
Would it be possible to add jsDoc (well tsdoc in fact) for this function ? We are building automatically a 'userguide' using a GitHub action like for this project for example: https://mljs.github.io/spectra-processing/

Reply:
I do not know how to write it, but I wite it in ReadMe.md